pyomo/contrib/mindtpy/util.py

```diff
@@ -318,3 +320,22 @@ def generate_norm1_norm_constraint(model, setpoint_model, config, discrete_only=
             for v_model, v_setpoint in zip(model_vars, setpoint_vars))
     norm_constraint_blk.sum_slack = Constraint(
         expr=sum(norm_constraint_blk.L1_slack_var[idx] for idx in norm_constraint_blk.L1_slack_idx) <= rhs)
+
+
+class IncumbentCallback_cplex(IncumbentCallback):
+    """Inherent class in Cplex to call Incumbent callback."""
+
+    def __call__(self):
+        """
+        This is an inherent function in LazyConstraintCallback in cplex. 
+        This callback will be used after each new potential incumbent is found.
+        https://www.ibm.com/support/knowledgecenter/SSSA5P_12.10.0/ilog.odms.cplex.help/refpythoncplex/html/cplex.callbacks.IncumbentCallback-class.html
+        """
+        solve_data = self.solve_data
+        opt = self.opt
+        integer_var_value_list = []
+        for var in solve_data.mip.MindtPy_utils.discrete_variable_list:
+            integer_var_value_list.append(round(self.get_values(opt._pyomo_var_to_solver_var_map[var])))
+        integer_var_value_tuple = tuple(integer_var_value_list)
+        if integer_var_value_tuple in solve_data.tabu_list:
+            self.reject()
```

Comment:
This looks WAY simpler than what I thought. I think one key issue might be performance with large problems, so there should be an option to write a temporary file with those solutions.

Reply:
Actually, I thought about this. So I use `set()` instead of `list()` here. I don't see the point to write a temporary file since we still need to read it, check if it's in the `list` or `set`, then write it. 
Do you mean writing the final tabu_list?

Reply:
I think that having the last tabu list might be useful (at least printing it in DEBUG logging) and that should solve the issue. The good part about writing the file is that you don't need to keep in memory `solve_data.tabu_list`, but you are right, we can live with that.